pkg/utils/stack_functions.go

```diff
@@ -0,0 +1,484 @@
+// the docker config data or the credential store.
+func resolveDockerConfRegKey(imgRegistry string) string {
+	var key string
+	switch imgRegistry {
+	// Docker registry: When logging in to the docker registry, the server name can be either:
+	// nothing, docker.io, index.docker.io, or registry-1.docker.io.
+	// They are all translated to: https://index.docker.io/v1/ as the server name.
+	case registry.IndexName, registry.IndexHostname, registry.DefaultV2Registry.Hostname():
+		key = registry.IndexServer
+	default:
+		key = imgRegistry
+	}
+
+	return key
+}
+
+func getClientClient() (client.Client, error) {
+	cl, err := client.New(config.GetConfigOrDie(), client.Options{})
+	if err != nil {
+		return nil, err
+	}
+	// if err != nil {
+	// 	panic(err.Error())
+	// }
+	return cl, err
+}
+
+func getKabanero() (kabanerov1alpha2.Kabanero, error) {
+
+	var kabaneroInstance kabanerov1alpha2.Kabanero
+	kabUnstructured := &unstructured.Unstructured{}
+	kabUnstructured.SetKind("Kabanero")
+	kabUnstructured.SetGroupVersionKind(schema.GroupVersionKind{
+		Kind:    "Kabanero",
+		Group:   kabanerov1alpha2.SchemeGroupVersion.Group,
+		Version: kabanerov1alpha2.SchemeGroupVersion.Version,
+	})
+
+	ctx := context.Background()
+	var err error
+	var cl client.Client
+	cl, err = getClientClient()
+	if err != nil {
+		return kabaneroInstance, err
+	}
+	ns := getHookNamespace()
+	n := types.NamespacedName{}
+	n.Name = ns
+	n.Namespace = ns
+	err = cl.Get(ctx, n, kabUnstructured)
+
+	kabaneroBytes, err := kabUnstructured.MarshalJSON()
+	if err != nil {
+		return kabaneroInstance, err
+	}
+
+	json.Unmarshal(kabaneroBytes, &kabaneroInstance)
+
+	return kabaneroInstance, err
+}
+
+func getStacks() (*unstructured.UnstructuredList, error) {
+	stacksUnstructured := &unstructured.UnstructuredList{}
+	stacksUnstructured.SetKind("Stack")
+	stacksUnstructured.SetGroupVersionKind(schema.GroupVersionKind{
+		Kind:    "Stack",
+		Group:   kabanerov1alpha2.SchemeGroupVersion.Group,
+		Version: kabanerov1alpha2.SchemeGroupVersion.Version,
+	})
+
+	ctx := context.Background()
+
+	var err error
+	var cl client.Client
+	cl, err = getClientClient()
+	if err != nil {
+		return stacksUnstructured, err
+	}
+	ns := getHookNamespace()
+	err = cl.List(ctx, stacksUnstructured, client.InNamespace(ns))
+
+	return stacksUnstructured, err
+}
+
+func getDeployments(name string, version string, ns string) ([]*models.DescribeStackAppsItems0, error) {
+	fmt.Println("entering deployments")
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		return nil, err
+	}
+	// if err != nil {
+	// 	panic(err.Error())
+	// }
+
+	// creates the clientset
+	clSet, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+	// if err != nil {
+	// 	panic(err.Error())
+	// }
+
+	fmt.Println("namespace")
+	fmt.Println(ns)
+	apps := []*models.DescribeStackAppsItems0{}
+	deploymentsClient := clSet.AppsV1().Deployments(ns)
```

Comment:
There's nothing wrong with using clientset here, but for the kab operator we consolidated everything to use the controller-runtime client.  It was easier to maintain.  Just a thought.

Reply:
for now, I'm going to stay with client set here